hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -176,8 +177,10 @@ protected HoodieTableMetaClient(HoodieStorage storage, String basePath, boolean
       checkArgument(layoutVersion.get().compareTo(tableConfigVersion.get()) >= 0,
           "Layout Version defined in hoodie properties has higher version (" + tableConfigVersion.get()
               + ") than the one passed in config (" + layoutVersion.get() + ")");
+    } else if (layoutVersion.isEmpty() && tableConfigVersion.isEmpty()) {
+      throw new TableNotFoundException("Table does not exist");
     }
-    this.timelineLayoutVersion = layoutVersion.orElseGet(() -> tableConfig.getTimelineLayoutVersion().get());
+    this.timelineLayoutVersion = layoutVersion.orElseGet(tableConfigVersion::get);
```

Comment:
I think the issue is that when `reloadTableConfig` is called then timelineLayoutVersion is not updated. We should load timelineLayoutversion within reloadTableConig.
```
  public synchronized void reloadTableConfig() {
    this.tableConfig = new HoodieTableConfig(this.storage, metaPath,
        this.tableConfig.getRecordMergeMode(), this.tableConfig.getKeyGeneratorClassName(), this.tableConfig.getRecordMergeStrategyId());
    reloadTimelineLayout();
  }
  
  private void reloadTimelineLayout() {
    this.timelineLayoutVersion = tableConfig.getTimelineLayoutVersion().get();
    this.timelineLayout = TimelineLayout.fromVersion(timelineLayoutVersion);
    this.timelinePath = timelineLayout.getTimelinePath(this.basePath);
  }
```

Reply:
take a look at the stack trace from the jira ticket

Reply:
We can discuss tomorrow if your mind isn't changed